### ISSUE TITLE
FileSystemException: Cannot open file client.js

### ISSUE THREAD

**Issue by sgehrman** [dart-lang/webdev#968]
building for web, I'm on beta channel.

FileSystemException: Cannot open file, path = '/home/steve/.pub-cache/hosted/pub.dartlang.org/dwds-3.0.2/lib/src/injected/client.js' (OS Error: No such file or directory, errno = 2)

**Comment by jakemac53** [comment]
Can you try doing a `pub cache repair` or possibly just deleting the directory for that version from your pub cache and re-running pub get? It sounds like maybe it didn't properly download.

**Comment by belolap** [comment]
Have same problem.

1. I remove directory `~/.pub-cache`
2. Reinstall flutter used beta channel
3. Reinstall VSCode extensions for dart & flutter
4. Make `flutter pub update` in my project's folder

And dwds is not restored in my `~/.pub-cache`.

**Comment by jakemac53** [comment]
Interesting, cc @jonahwilliams

Is this usually downloaded as a part of creating the flutter_tool snapshot? (does that run pub get in the flutter_tool dir?)

If so it sounds like after that point if somebody cleans their pub cache the tool still relies on the files being there but they will not be restored. I am not totally sure what the best option here is.

Possibly flutter_tool could attempt to validate its own `package_config.json` file and ensure all the dirs listed there exist?

**Comment by jonahwilliams** [comment]
Yeah, there are a few ways to validate this but I think the only way we could fix this from a snapshot is to force it to be regenerated with a new package_config.json. A different approach (in flutter's case) would be to allow the tooling to provide the bytes directly instead of using resolvePackageUri. If the tool failed to find it with package resolution, we could run pub get and load the bytes from a known location.

**Comment by jakemac53** [comment]
I think that dwds does just expose a handler now, which you could precede with your own handler to serve this file from wherever you wish (you could copy it into some other more permanent location and not rely on the pub cache for instance).
